From: wuyifeng <wuyifeng@nd.com.cn>
Date: Thu, 19 Jan 2023 09:46:01 +0800
Subject: Revert Enable-StrictOriginIsolation-and-SitePerProcess.patch

Fix version: 1.4.0
The 'Enable-StrictOriginIsolation-and-SitePerProcess.patch' will cause the issue that Drag
and Drop function doesn't work, so revert it.
Relate to https://prometheanworld.atlassian.net/browse/AP9-11548.
Relate to https://prometheanworld.atlassian.net/wiki/spaces/AP9/pages/14429487423/SPIKE+-+Drag+and+Drop+function+in+Chromium+Browser+does+not+work.
Relate to Revert-Enable-SPPI-for-devices-with-enough-memory.patch, Disable-transferring-link-and-image-of-Drag-Drop-function.patch.

---
 chrome/browser/about_flags.cc                      | 3 +++
 chrome/browser/chrome_content_browser_client.cc    | 4 ++--
 components/site_isolation/site_isolation_policy.cc | 2 --
 content/public/common/content_features.cc          | 4 ++--
 4 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -4596,6 +4596,9 @@ const FeatureEntry kFeatureEntries[] = {
      flag_descriptions::kSiteIsolationForPasswordSitesDescription, kOsAndroid,
      FEATURE_VALUE_TYPE(
          site_isolation::features::kSiteIsolationForPasswordSites)},
+    {"enable-site-per-process", flag_descriptions::kStrictSiteIsolationName,
+     flag_descriptions::kStrictSiteIsolationDescription, kOsAndroid,
+     SINGLE_VALUE_TYPE(switches::kSitePerProcess)},
 #endif
     {"enable-isolated-web-apps", flag_descriptions::kEnableIsolatedWebAppsName,
      flag_descriptions::kEnableIsolatedWebAppsDescription, kOsAll,
diff --git a/chrome/browser/chrome_content_browser_client.cc b/chrome/browser/chrome_content_browser_client.cc
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -1425,7 +1425,7 @@ void ChromeContentBrowserClient::RegisterLocalStatePrefs(
   registry->RegisterFilePathPref(prefs::kDiskCacheDir, base::FilePath());
   registry->RegisterIntegerPref(prefs::kDiskCacheSize, 0);
   registry->RegisterStringPref(prefs::kIsolateOrigins, std::string());
-  registry->RegisterBooleanPref(prefs::kSitePerProcess, true);
+  registry->RegisterBooleanPref(prefs::kSitePerProcess, false);
   registry->RegisterBooleanPref(prefs::kTabFreezingEnabled, true);
   registry->RegisterIntegerPref(prefs::kSCTAuditingHashdanceReportCount, 0);
 }
@@ -1439,7 +1439,7 @@ void ChromeContentBrowserClient::RegisterProfilePrefs(
   // user policy in addition to the same named ones in Local State (which are
   // used for mapping the command-line flags).
   registry->RegisterStringPref(prefs::kIsolateOrigins, std::string());
-  registry->RegisterBooleanPref(prefs::kSitePerProcess, true);
+  registry->RegisterBooleanPref(prefs::kSitePerProcess, false);
   registry->RegisterListPref(
       site_isolation::prefs::kUserTriggeredIsolatedOrigins);
   registry->RegisterDictionaryPref(
diff --git a/components/site_isolation/site_isolation_policy.cc b/components/site_isolation/site_isolation_policy.cc
--- a/components/site_isolation/site_isolation_policy.cc
+++ b/components/site_isolation/site_isolation_policy.cc
@@ -169,7 +169,6 @@ bool SiteIsolationPolicy::IsIsolationForOAuthSitesEnabled() {
 
 // static
 bool SiteIsolationPolicy::IsEnterprisePolicyApplicable() {
-  if ((true)) return true;
 #if BUILDFLAG(IS_ANDROID)
   // https://crbug.com/844118: Limiting policy to devices with > 1GB RAM.
   // Using 1077 rather than 1024 because 1) it helps ensure that devices with
@@ -185,7 +184,6 @@ bool SiteIsolationPolicy::IsEnterprisePolicyApplicable() {
 // static
 bool SiteIsolationPolicy::ShouldDisableSiteIsolationDueToMemoryThreshold(
     content::SiteIsolationMode site_isolation_mode) {
-  if ((true)) return false;
   static const bool cache_memory_threshold_decision =
       ShouldCacheMemoryThresholdDecision();
   if (!g_disallow_memory_threshold_caching && cache_memory_threshold_decision) {
diff --git a/content/public/common/content_features.cc b/content/public/common/content_features.cc
--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -1116,8 +1116,8 @@ BASE_FEATURE(kStopVideoCaptureOnScreenLock,
 // Controls whether site isolation should use origins instead of scheme and
 // eTLD+1.
 BASE_FEATURE(kStrictOriginIsolation,
-             "StrictOriginIsolation",              // must be enabled
-             base::FEATURE_ENABLED_BY_DEFAULT);    // in Bromite
+             "StrictOriginIsolation",
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Disallows window.{alert, prompt, confirm} if triggered inside a subframe that
 // is not same origin with the main frame.
-- 
2.34.1

